datafusion/physical-expr/src/expressions/datetime.rs

```diff
@@ -43,6 +44,7 @@ pub struct DateTimeIntervalExpr {
     lhs: Arc<dyn PhysicalExpr>,
     op: Operator,
     rhs: Arc<dyn PhysicalExpr>,
+    input_schema: Schema,
```

Comment:
```suggestion
    // TODO: move type checking to the planning phase and not in the physical expr 
    // so we can remove this
    input_schema: Schema,
```

This is unfortunate -- I see you need to do it so you can call `try_new()` -- but really I think DataFusion should have done this type checking in the planning phase rather than at the physical expr stage -- maybe I'll file a ticket to improve that.